roachtest: inc wait to 10m for token return in perturbation tests

andrewbaptist opened the pull request:
We assert that tokens are returned at the end of the `perturbation/*`
roachtests. However, we only waited 5 seconds before failing the
assertion. As the comment in this patch suggests, draining a send queue
can take longer as the node paces its catchup. In such cases, it is
acceptable that tokens are not returned within the lesser duration.

Bump this duration to 10 minutes.

Fixes: #136297
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/135211)
<!-- Reviewable:end -->

kvoli commented:
It may need to be even longer, judging by the partition run. Have you validated the 5 min duration is sufficient?

andrewbaptist commented:
No - I can set this even longer, let me test this out, so hold off reviewing until I can run a couple of successful runs against it.

kvoli commented:
> Is this ready for review @andrewbaptist?

bump @andrewbaptist

kvoli commented:
Andrew is OOO. I'm going to bump this to 10 minutes and then merge. We hit this on https://github.com/cockroachdb/cockroach/pull/136253 so it is therefore on the critical path to enabling `apply_to_all` on master.

kvoli committed:
roachtest: inc wait to 10m for token return in perturbation tests

We assert that tokens are returned at the end of the `perturbation/*`
roachtests. However, we only waited 5 seconds before failing the
assertion. As the comment in this patch suggests, draining a send queue
can take longer as the node paces its catchup. In such cases, it is
acceptable that tokens are not returned within the lesser duration.

Bump this duration to 10 minutes.

Fixes: #136297
Release note: None

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> though we could just drop the first commit entirely.

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135211#-:-OCiBRl68DvdZ5yPHoNO:b-3iuvxe)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @herkolategan, @kvoli, @srosenberg, and @sumeerbhola)



<!-- Sent from Reviewable.io -->

kvoli reviewed (commented):
Good idea, done. TYFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/135211#-:-OCiCJq89zsSZxLPqk4s:blhcdkv)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @srosenberg, and @sumeerbhola)



<!-- Sent from Reviewable.io -->

kvoli commented:
bors r=arulajmani

andrewbaptist commented:
blathers backport 24.3
